### PR TITLE
Adding the sed replacement sd, written in rust

### PR DESCRIPTION
**Pull request by abhi18av**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by etu** [comment]
Without actually looking at the code, you would need to put the file in it's own directory and not just as `default.nix` in the misc directory. I haven't made up my mind if misc is the right place, but it probably is. But at least it needs to be in a directory named `sd` in there.

Secondly you need to add a line to the `all-packages.nix` to reference the derivation.

And thirdly you should amend your commit and push it with a message like: `sd: init at 0.4.3` to follow the contributing guidelines for commit messages.

**Comment by abhi18av** [comment]
I see, thanks @etu 

I've added myself as a maintainer and I'll be re-working this PR soon.

**Comment by Shou** [comment]
I wonder if `pkgs/tools/text` would be a better place? That's where `ripgrep`, `gnused`, `the_silver_searcher` is.

File: pkgs/tools/misc/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+, darwin, libiconv
+}:
+
+with rustPlatform;
+
+buildRustPackage rec {
+  name = "sd-${version}";
+  version = "0.4.3";
+
+  cargoSha256 = "08zzn3a32xfjkmpawcjppn1mr26ws3iv40cckiz8ldz4qc8y9gdh";
```
Comment (liff) [inline pkgs/tools/misc/default.nix]: This hash is from the `exa` package. Should be changed to the correct value that you can get from a failed build, like [described in the manual](https://nixos.org/nixpkgs/manual/#compiling-rust-applications-with-cargo).

File: pkgs/tools/misc/default.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  src = fetchFromGitHub {
+    owner = "chmln";
+    repo = "sd";
+    rev = "v${version}";
+    sha256 = "0jy11a3xfnfnmyw1kjmv4ffavhijs8c940kw24vafklnacx5n88m";
```
Comment (liff) [inline pkgs/tools/misc/default.nix]: This hash is also from the `exa` package. You should be able to get the correct hash with this command:

```bash
nix run nixpkgs.nix-prefetch-git -c nix-prefetch-git https://github.com/chmln/sd 0.4.3
```

File: pkgs/tools/misc/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  };
+
+  nativeBuildInputs = [ cmake pkgconfig perl ];
+  buildInputs = [ zlib ]
```
Comment (liff) [inline pkgs/tools/misc/default.nix]: Neither of `nativeBuildInputs` or `buildInputs` are needed for sd.

File: pkgs/tools/misc/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  ++ stdenv.lib.optionals stdenv.isDarwin [
+    libiconv darwin.apple_sdk.frameworks.Security ]
+  ;
+
+  postInstall = ''
```
Comment (liff) [inline pkgs/tools/misc/default.nix]: This is also from `exa` and should be removed.

File: pkgs/tools/misc/default.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    mkdir -p $out/share/fish/vendor_completions.d
+    cp contrib/completions.fish $out/share/fish/vendor_completions.d/exa.fish
+
+    mkdir -p $out/share/zsh/site-functions
+    cp contrib/completions.zsh $out/share/zsh/site-functions/_exa
+  '';
+
+  # Some tests fail, but Travis ensures a proper build
+  doCheck = false;
```
Comment (liff) [inline pkgs/tools/misc/default.nix]: Not needed for sd.

File: pkgs/tools/misc/default.nix [hunk 6]
```diff
@@ -0,0 +1,52 @@
+{ stdenv, fetchFromGitHub, rustPlatform, cmake, perl, pkgconfig, zlib
+, darwin, libiconv
+}:
+
+with rustPlatform;
+
+buildRustPackage rec {
+  name = "sd-${version}";
```
Comment (liff) [inline pkgs/tools/misc/default.nix]: Could use `pname = "sd"` instead.

**Comment by dywedir** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/60241

**Pull request closed without merging**